cpp/CMakeLists.txt

```diff
@@ -61,9 +61,32 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     endif(CMAKE_CXX11_ABI)
 endif(CMAKE_COMPILER_IS_GNUCXX)
 
-#set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_60,code=sm_60 -gencode=arch=compute_61,code=sm_61")
-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_60,code=sm_60")
-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_70,code=compute_70")
+# Auto-detect available GPU compute architectures
+set(GPU_ARCHS "ALL" CACHE STRING
+  "List of GPU architectures (semicolon-separated) to be compiled for. Pass 'ALL' if you want to compile for all supported GPU architectures. Empty string means to auto-detect the GPUs on the current system")
```

Comment:
Since `ALL` is the default, maybe we want to use `AUTO` instead of an empty string for the respective option?

Reply:
An empty string is what cuML uses, so I kept it the same in case someone's using the same cmake arguments to build both. Not sure how many people will be doing that besides myself though.

Reply:
Fair enough, I just imagine someone inadvertently deleting `GPU_ARCHS=` out of their `.env` file, assuming it was a no-op.

Speaking of myself, of course. :)